frappe/core/doctype/role_profile/role_profile.py

```diff
@@ -34,24 +34,16 @@ def on_update(self):
 
 	def update_all_users(self):
 		"""Changes in role_profile reflected across all its user"""
-		has_role = frappe.qb.DocType("Has Role")
-		user = frappe.qb.DocType("User")
-
-		all_current_roles = (
-			frappe.qb.from_(user)
-			.join(has_role)
-			.on(user.name == has_role.parent)
-			.where(user.role_profile_name == self.name)
-			.select(user.name, has_role.role)
-		).run()
-
-		user_roles = defaultdict(set)
-		for user, role in all_current_roles:
-			user_roles[user].add(role)
-
-		role_profile_roles = {role.role for role in self.roles}
-		for user, roles in user_roles.items():
-			if roles != role_profile_roles:
-				user = frappe.get_doc("User", user)
-				user.roles = []
-				user.add_roles(*role_profile_roles)
+		users = frappe.get_list("User Role Profile", filters={"role_profile": self.name}, pluck="parent")
+		for user in users:
+			role_profile_roles = []
+			user = frappe.get_doc("User", user)
```

Comment:
This will be quite slow :sweat_smile: not sure about alternate implementation yet but this is gonna become a problem some day for sure.

Reply:
Optimization: 
- Get users with this role profile, also get all of their role profiles
- Reconstruct new roles that they'll have. If they are same then odn't reupate, else update.
- No need to write any business logic here. user.py is already doing it on save?

Reply:
We can optimize it further as you suggested, at least this way we will have advantage in comparison with join queries. 😅 

> No need to write any business logic here. user.py is already doing it on save?

Previously also user.py did not have the logic to update the user, if profile is itself update.  🤔

Reply:
@niraj2477 previous implementation avoided saving user if there was no change, here we are re-saving anyway so everything will get triggered. 

https://github.com/frappe/frappe/pull/24745/commits/9c2c3112a036cdb7ae1c2f56b2bb662d745ee37b